proxy/proxy.go

```diff
@@ -966,6 +966,12 @@ func (p *Proxy) makeUpgradeRequest(ctx *context, req *http.Request) {
 }
 
 func (p *Proxy) makeBackendRequest(ctx *context, requestContext stdlibcontext.Context) (*http.Response, *proxyError) {
+	var payloadProtocol string
+	isUpgrade := isUpgradeRequest(ctx.Request())
+	if isUpgrade {
+		payloadProtocol = ctx.Request().Header.Get("Upgrade")
```

Comment:
I think there is getUpgradeRequest helper for this.

Reply:
ah yeah makes sense to use. At first I thought we don't need the value and then I forgot the func.